docs: blog post, migration guide and roadmap (#228)
siemens/ix-docs: docs/home/releases/roadmap.md

![Roadmap](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=6992-210&t=UmcDJUatVbcLymkF-4)

### ✅ Completed

> **🔹 Release 5.0**: Applying aligned color reference tokens, refactoring keyboard navigation in dropdowns, new [date-time input](/docs/components/input-date-time) & [range field](/docs/components/range-field) component, [tabs](/docs/components/tabs) with overflow menu and style adaptions. [Read more](/blog/2026/05/11/release-5)

Reviewer: aligned with what?  other systems?

Author: Aligned with Siemens design language

> ### 🚧 In progress / Currently being worked on

Reviewer: do we need both headings? "Work in progress" is probably better

> - **Popover**: The new popover component bridges the gap between small text-based tooltips and modal dialogs. In contrast to an on-hover tooltip the popover typically opens on-click. It is built with great flexibility providing slot options for header, content and footer.

Reviewer: ```suggestion
- **Popover**: The new popover component bridges the gap between small text-based tooltips and modal dialogs. In contrast to an on-hover tooltip, the popover typically opens upon clicking. It is built with great flexibility providing slot options for header, content and footer.
```

> - **Badge**: The badge component is closely related to the current pill component. It is always bound to an anchor element and can be used to indicate notifications, showing a count or conveying status.

Reviewer: "The new badge component..." ???? If new.

- **Inline notification**: The inline notification offers an alternative to larger message bars. Inline notifications appear within a specific section or component of a page, rather than spanning the entire width. 

**🔹 Major release 6.0**: Planned for Q4 2026, see info on [release strategy](release-version.md).

> ### 🗓️ Scheduled / With tentative timeline

Reviewer: ```suggestion
### 🗓️ Scheduled
```

*LOL again not sure both headings are required

- **Generic list**: The generic list extends our component set with a more flexible solution for lists. It comes as a customizable variant providing a slot and a pre-defined variant with a useful default of checkbox selection, icon, title and actions.
- **Building blocks**: Blocks are curated, copy‑paste code examples that show how multiple components work together in real UI scenarios. Blocks bridge the gap between individual components and larger patterns, giving users practical, ready‑to-use starting points.

> - **Vertical navigation menu rework**: Our vertical navigation will get a facelift in alignment with other design systems in the Siemens organization.

Reviewer: ```suggestion
- **Vertical navigation menu rework**: Our vertical navigation will change in alignment with other design systems at Siemens.
```

> - **Settings and about as modal**: Currently, settings and the about information are presented in an overlay that is bound to the vertical navigation menu. As a more flexible version we will provide a template for settings and about information in a modal that can either come at a fixed width or full-screen. Access to this modal will move from the bottom left menu navigation to the header.

Reviewer: ```suggestion
- **Settings and about as modal**: Currently, settings and the about information are presented in an overlay that is bound to the vertical navigation menu. For a more flexible version, we will provide a template for settings and about information in a modal that can either come at a fixed width or full-screen. Access to this modal will move from the bottom left menu navigation to the header.
```

> ### ⏳ Planned / Without specific timeline

Reviewer: again 2 headings not necessary

> - **Density options**: Density tokens allow changing sizes and spacings of elements on the screen without creating different variants of components. It uses the same mechanism as switching color modes or themes. To take advantage of the density modes, all relevant dimensions of a component must use density tokens, e.g. for font size and line height, widths, heights, paddings, and gaps.

Reviewer: ```suggestion
- **Density options**: Density tokens allow for changing sizes and spacings of elements on the screen without creating different variants of components. It uses the same mechanism as switching color modes or themes. To use density modes, all relevant dimensions of a component must use density tokens, e.g. for font size and line height, widths, heights, paddings and gaps.
```

> - **Box-styled tabs**: To grant more flexibility we are adding an option to our tabs component to appear in a box-style manner, containering the content below. This allows two visually distinct levels of tab navigation

Reviewer: ```suggestion
- **Box-styled tabs**: For more flexibility, we're adding an option to our tabs component to appear in a box-style manner, containing the content below. This allows two visually distinct levels of tab navigation
```
